### DISCUSSION TITLE
quarto render log output without buffer?

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#8467] [Q&A] [answered]
### Description

I have a typescript script that I use to do several pre-pre render operations and at the end run quarto render with

```typescript
    const process = Deno.run({
        cmd: ["quarto", "render"],
        stdout: "inherit",
        stderr: "inherit",
    });
```

When I run this script, it seems to buffer all of the console logs from the render process; only when and if the render completes does it show the console logs of each of the documents being rendered. I don't see anything in `quarto render --help` that toggles off this buffering - is this just not possible atm?

Being able to switch off the buffering would be useful for debugging: if the script hangs during the render process, there's no log output to know which doc was the offender.

**Comment by mcanouil** [comment]
CC @cscheid as my TypeScript knowledge is pretty basic (especially in the context of Quarto processing)

**Reply by cscheid** [reply]
... is this a quarto question or a Deno question? If you run quarto from a shell and redirect the output to stderr, do you still get this behavror?

**Reply by andrewpbray** [reply]
I've worked a bit on reconfiguring how the command is called and how stderr is handled but no dice so far. I also have fairly basic Typescript knowledge. I'll keep working on it and either post a solution or a reprex containing the approaches that haven't worked.

**Reply by cscheid** [reply]
> I've worked a bit on reconfiguring how the command is called and how stderr is handled but no dice so far.

Quarto does do a bit of wonky stream redirecting, so it's very plausible we could be going wrong. I suggested to run from a shell to try and isolate the behavior.

**Reply by andrewpbray** [reply, marked as answer]
I upgraded from 1.3 to 1.4 and the issue is resolved! With no change to the script or the options uses in `Deno.run`, it now doesn't hold the render console logs in a buffer.
